### PR TITLE
Add train step and data loading

### PR DESCRIPTION
**Pull request by bytehut**
Notes:
- Make sure to update the foldername at the top of LSTM.ipynb when you try to compile.
- Notebook compiles and all cells run without errors.

Current issues:
- Loss shows up as 0 for all epochs. Note that we currently don't use the same loss fcn as Alec. Possible that the data is not loaded properly or some other issue leading up to training.

### CODE REVIEW

File: GuitarLSTM/LSTM.ipynb [hunk 1]
```diff
@@ -1,67 +1,235 @@
...
+    "      # Initialise network hidden state by processing some samples then zero the gradient buffers\n",
+    "      self(input_batch[:, 0:init_len, :])\n",
+    "      self.zero_grad()\n",
+    "\n",
+    "      # Choose the starting index for processing the rest of the batch sequence, in chunks of args.up_fr\n",
+    "      start_i = init_len\n",
+    "      batch_loss = 0\n",
+    "      k = 0\n",
+    "      # Iterate over the remaining samples in the mini batch\n",
+    "      for k in range(math.ceil((input_batch.shape[0] - init_len) / up_fr)):\n",
```
Comment (bradyhhhuang) [inline GuitarLSTM/LSTM.ipynb]: the reason why the loss is 0 is because math.ceil((input_batch.shape[0] - init_len) / up_fr) is 0 .  batch_size - inin_len<0 when batch_size=10 and init_len=200. When I set batch_size =1000, there's a loss

Comment (bytehut) [reply GuitarLSTM/LSTM.ipynb]: Oh I see the issue, this should be "input_batch.shape[1]". Nice catch!

**Review by bradyhhhuang — APPROVED** [review]
